src/core/qgsvectorlayerfeatureiterator.cpp

```diff
@@ -530,7 +530,8 @@ bool QgsVectorLayerFeatureIterator::isValid() const
 
 bool QgsVectorLayerFeatureIterator::fetchNextAddedFeature( QgsFeature &f )
 {
-  while ( mFetchAddedFeaturesIt-- != mSource->mAddedFeatures.constBegin() )
+  while ( mFetchAddedFeaturesIt != mSource->mAddedFeatures.constBegin() &&
```

Comment:
This is really quite confusing to read. The mFetchAddedFeaturesIt-- != mSource->mAddedFeatures.constBegin() test seems to be useless to me and, as a test, be equivalent to mFetchAddedFeaturesIt != mSource->mAddedFeatures.constBegin() 

Wouldn't the following be better
```
while( mFetchAddedFeaturesIt != mSource->mAddedFeatures.constBegin() )
{
    --mFetchAddedFeaturesIt;
 ```

Reply:
yep, i'll fix